content/client-lib-development-guide/features.textile

```diff
@@ -152,7 +150,7 @@ h3(#rest-channel). Channel
 ** @(RSL1c)@ When an array of @Message@ objects is provided, a single request is made to Ably. When publishing multiple messages, this approach is more efficient. However, a yet to be implemented feature should limit the total number of messages bundled in a single POST based on the default max request size, and would raise an exception if any single message exceeds that limit.
 ** @(RSL1d)@ Indicates an error if the message was not successfully published to Ably
 ** @(RSL1e)@ Allows @name@ and or @data@ to be @null@.  If any of the values are null, then key is not sent to Ably i.e. a payload with a @null@ value for @data@ would be sent as follows @{ "name": "click" }@
-** @(RSL1f)@ Unauthenticated clients without a @clientId@ (any @clientId@ is permitted):
+** @(RSL1f)@ Unauthenticated clients using basic auth without a @clientId@ (i.e. any @clientId@ is permitted):
```

Comment:
> We could revert to explicitly requiring a wildcard? And an empty clientId means anonymous?

Well there is one other option, which is to not allow a `clientId` if `clientId` is null / undefined in the TokenParams, and if `clientId` is an empty string, then any `clientId` is allowed.  The reasoning behind that is because an empty string `clientId` is not allowed so there is no possibility of a `clientId` with an empty string ever colliding.  However, I don't think that is particularly user friendly either.  I would perhaps go with no `clientId` means the user is not authenticated, and `*` is any `clientId` is allowed.  It's the most intuitive, and we're only then making one `clientId` impossible to specify in TokenParams, that being `*`.  Hardly a big problem I would say.